transifex: Setup Transifex for better translation process.

manavmehta opened the pull request:
**What's this PR do?**

1. add .tx/config to configure communication with Transifex CLI
2. add how-to-doc for translating zulip-desktop
3. add tx-pull executable script for pulling translations from Transifex
4. update supported-locales.json
5. remove now-inessential scripts and translations

and other necessary changes

**You have tested this PR on:**
  - [X] macOS
  - [ ] Ubuntu 20.04

manavmehta committed:
transifex: Setup Transifex - initial commit

1. add .tx/config to configure communication with transifex CLI
2. add how-to-doc for translating zulip-desktop
3. add tx-pull executable script for pulling translations from transifex

manavmehta commented on docs/howto/translations.md:
```diff
@@ -0,0 +1,95 @@
+three boring, more-or-less mechanical lists:
+ * `translations/flow-typed/translations.js`
+ * `translations/i18n/locale.js`
+ * `translations/i18n/messages.js`
```
These might or might not be required as-it-is and are prone to changes

timabbott commented:
The slug in `.tx/config` should be: `zulip.desktopjson`

Otherwise, the Transifex integration wfm.

timabbott committed:
Update translation data from Transifex.

timabbott commented:
I fixed that slug, ran `tools/tx-pull`, and pushed back here following that change.

manavmehta commented:
@akashnimare FYI.
May we look into removing the inessential modules [here](https://github.com/zulip/zulip-desktop/tree/master/tools/locale-helper)?

akashnimare commented:
@manavmehta yes, please remove that since we won't be using that module.

manavmehta committed:
revert inessential parts of 8e04920

As now we won't be needing the google-translate-api, we need to remove the redundant dependency,
and the locale-helper directory which may override our Transifex translations

We will preserve the translated key-value json modules and the executable scripts from the commit

manavmehta commented:
@akashnimare Removed the locale-helper from 8e04920 which is now not needed.
I've updated the `supported-locales` module :)
I've used native scripts for language names as we use them on the webapp. If you feel we could go for English names only.
cc: @timabbott

manavmehta committed:
Update supported-locales.json and other necessary changes

1. Update supported-locales.json to accommodate new languages added (with native scripts)
2. Remove en-US as it is not on our Transifex repository
3. Set default dropdown language as en, as a result of *2.*

akashnimare commented:
`tx-pull` works fine. @andersk can you give a quick review so that we can merge it?

akashnimare commented:
This is GTM after we pull a few strings which are there on Trasifex.

akashnimare commented:
Merging. Will update the strings on tx.
